File: engine/engine-model/src/main/scala/tools/packet/foreigneffect/ShowForeignInfo.scala
package tools.packet.foreigneffect

import net.opcodes.SendOpcode
import tools.packet.PacketInput

class ShowForeignInfo(private var _characterId: Int, private var _path: String) extends PacketInput {
  def characterId: Int = _characterId

  def path: String = _path

  override def opcode(): SendOpcode = SendOpcode.SHOW_FOREIGN_EFFECT
}

File: engine/engine-model/src/main/java/tools/packet/foreigneffect/ShowForeignInfo.java
package tools.packet.foreigneffect;

import net.opcodes.SendOpcode;
import tools.packet.PacketInput;

public record ShowForeignInfo(Integer characterId, String path) implements PacketInput {
   @Override
   public SendOpcode opcode() {
      return SendOpcode.SHOW_FOREIGN_EFFECT;
   }
}